### DISCUSSION TITLE
Spreading operations evenly between resources

### DISCUSSION THREAD

**Discussion by tjheiska** [frePPLe/frepple#439] [Q&A]
Hello,

Is it possible to schedule the operations equally between operation resources? I have tried to change the search mode for operationresource to variate things, but it seems that everything is scheduled for the first resource option, if the item is possible to make on time.

What I really would like to do is to have the tasks scheduled as tight as possible using all available resources and not mind them being early.

Thanks!

**Comment by jdetaeye** [comment]
A more balanced usage of your resources isn't too difficult to configure:  a) Use "minpenalty" als search mode, and b) assure the resource cost field is different from 0.

**Reply by tjheiska** [reply]
Thanks for quick answer again. I already had the MINPENALTY for my operationresources. I now also changed the cost field for my parent resource from 0 -> 1, but seems all tasks are still scheduled for the first resource and other two are left without jobs.

**Comment by jdetaeye** [comment]
Try setting the parameter plan.loglevel to 2 and check out how the algorithm evaluates each of the alternatives.

**Reply by tjheiska** [reply]
Ok, did that. The logs showed my cost as 0 for every resource, so moved the cost from parent resource to individual children. But penalty seems to be zero most of the time for every choice. And that makes the algorithm to pick the first alternative. So I still have 20 jobs for first operator, 4 jobs for second and nothing for third.
Any more tips or ideas?

**Reply by jdetaeye** [reply]
>>What I really would like to do is to have the tasks scheduled as tight as possible using all available resources and not mind them being early.

Rereading your description it looks like you want an ASAP plan.   In addition to the above, try setting the parameter "plan.administrativeleadtime" to 100 (days).

**Reply by tjheiska** [reply]
ASAP-plan sounds good. After making the changes, tasks seem to spread little nicer and gaps are definitely smaller. Unfortunately I still have challenges with this. 

In my model, producing an item needs 4 work stages.  Second stage is the most crowded one and that seems to be planned tight and evenly, but previous or following stages are not that nicely scheduled. Probably there's no way to schedule the following stages as nice, since they depend on the crowded stage. But maybe there's some way to change things to get the first stage to be scheduled more ASAP-way? 

In my current model I have every stage as its own operation producing an item consumed by the next stage. Maybe there's a way to make sale order of each stage separately (and have due date per stage) and somehow tell the system that  "Sale order A should be ready before starting to make sale order B"?

**Comment by jdetaeye** [comment]
>>In my model, producing an item needs 4 work stages. Second stage is the most crowded one and that seems to be planned tight and evenly, but previous or following stages are not that nicely scheduled. Probably there's no way to schedule the following stages as nice, since they depend on the crowded stage. But maybe there's some way to change things to get the first stage to be scheduled more ASAP-way?

You should read up on Theory Of Constraints
https://www.tocinstitute.org/theory-of-constraints.html
https://en.wikipedia.org/wiki/Theory_of_constraints

The plan should follow the pace of the bottleneck resource (what you call "most crowded one"). The rest of the plan should be centered around the bottleneck.
Planning all stages in ASAP mode is very bad practice.   It tends to push a lot of useless WIP/work in progress in the factory. Modern shop floor management a) prefer keeping WIP levels low&lean&agile, b) favor a smooth&simple flow of WIP through the factory and c) favor pull-based planning rather than push-based planning.

>>In my current model I have every stage as its own operation producing an item consumed by the next stage. Maybe there's a way to make sale order of each stage separately (and have due date per stage) and somehow tell the system that "Sale order A should be ready before starting to make sale order B"?

You can build in safety time in your schedule with the "post operation time" field of the operation.
The parameter "plan.administrativeleadtime" is a global way, while the "post operation time" field is a more granular way of controlling where you want to add safety time in your plan.
So, yes - you can go ahead and stuff the factory with all your ASAP work-in-progress inventory :-)

**Reply by tjheiska** [reply]
Well, thanks again. I most likely need to play a little more with these parameters and read some more theory. You're right that ASAP plan for all stages would be bad. But I'm trying to optimize that individual operator would have in sane amount of tasks when at workstation and longer gaps instead of lots of them.
